src/librustc/ty/sty.rs

```diff
@@ -385,7 +385,8 @@ impl<'tcx> ClosureSubsts<'tcx> {
         let ty = self.closure_sig_ty(def_id, tcx);
         match ty.sty {
             ty::FnPtr(sig) => sig,
-            _ => bug!("closure_sig_ty is not a fn-ptr: {:?}", ty),
+            ty::Infer(_) | ty::Error => ty::Binder::dummy(FnSig::fake()),  // ignore errors
```

Comment:
Looks like this change can be removed since it didn't end up being used in `suggest_fn_call`.

Reply:
(It contradicts the "only usable outside of an inference context" from the function description and can hide an error if it *is* called from an inference context.)